src/stakingrewards/stakingrewards.js

```diff
@@ -308,12 +308,15 @@ exports.getOngoingMekleInput = async function (
           ? parseInt(epochStake.epochDuration)
           : currentTime - epochStake.epochTimestamp
 
-        // If the operator was confirmed in the middle of this epoch...
         if (
+          // If the operator was confirmed in the middle of this epoch...
           opConfTimestamp > epochTimestamp &&
           opConfTimestamp <= epochTimestamp + epochDuration
         ) {
           epochDuration = epochTimestamp + epochDuration - opConfTimestamp
+        } else if (opConfTimestamp >= epochTimestamp + epochDuration) {
+          // No rewards if the operator was not yet confirmed
+          epochDuration = 0
```

Comment:
Minor nitpick and up to you if you want to make the change or not: to be clearer here we should name this variable (initialized on L307) to be `eligibleEpochDuration` or something similar.